data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -337,16 +360,27 @@ public void doOutput(final Collection<Record<Event>> records) {
       } catch (IOException | EventKeyNotFoundException e) {
           LOG.error("There was an exception when constructing the index name. Check the dlq if configured to see details about the affected Event: {}", e.getMessage());
           dynamicIndexDroppedEvents.increment();
-          logFailureForDlqObjects(List.of(DlqObject.builder()
-                  .withEventHandle(event.getEventHandle())
-                  .withFailedData(FailedDlqData.builder().withDocument(event.toJsonString()).withIndex(indexName).withMessage(e.getMessage()).build())
-                  .withPluginName(pluginSetting.getName())
-                  .withPipelineName(pluginSetting.getPipelineName())
-                  .withPluginId(pluginSetting.getName())
-                  .build()), e);
+          logFailureForDlqObjects(List.of(createDlqObjectFromEvent(event, indexName, e.getMessage())), e);
           continue;
       }
 
+      Long version = null;
```

Comment:
There is a lot of untested code here. We should have unit tests for these.

Are there opportunities to start splitting logic out of this class to help with testing?

Reply:
This entire class does not have a unit test, but sure I can start it to test this new code